runtime/src/main/java/org/corfudb/runtime/CheckpointWriter.java

```diff
@@ -244,36 +275,24 @@ private long nonCachedAppend(Object object, UUID ... streamIDs) {
      *
      * @return Stream of global log addresses of the CONTINUATION records written.
      */
-    public int appendObjectState(Stream<Map.Entry> entryStream) {
+    public long appendObjectState(Stream<Map.Entry> entryStream) {
         ImmutableMap<CheckpointEntry.CheckpointDictKey, String> mdkv =
                 ImmutableMap.copyOf(this.mdkv);
 
         final Iterable<List<Map.Entry>> partitions =
                 Iterables.partition(entryStream::iterator, batchSize);
-        int entryCount = 0;
 
+        ExecutorService executor = Executors.newFixedThreadPool(MAX_WRITER);
+
+        int i = 0;
         for (List<Map.Entry> partition : partitions) {
```

Comment:
FYI, I think the partitions are lazily evaluated, so for the RocksDB implementation, you need to create an iterator per writer thread and the iterators have to start at different positions (they have to partition the state) .

Reply:
Will look into the RocksDB issue in another patch.